internal/process/defunct_processes.go

```diff
@@ -1,59 +1,90 @@
 package process
 
 import (
-	"fmt"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
 	"os"
 	"strconv"
 	"strings"
 )
 
-// DefunctProcesses returns the PIDs of all the zombie processes in the node
-func DefunctProcesses() ([]int, error) {
-	directories, err := os.Open("/proc")
+// ProcessFS is the process file system.
+const ProcessFS = "/proc"
+
+// Stat represents status information of a process from /proc/[pid]/stat.
+type Stat struct {
+	// Fname is the filename of the executable.
+	Fname string
+
+	// State is the state of the process.
+	State string
+}
+
+// DefunctProcesses returns the number of zombie processes in the node.
+func DefunctProcesses() (int, error) {
+	directories, err := os.Open(ProcessFS)
 	if err != nil {
-		return nil, err
+		return -1, err
```

Comment:
@saschagrunert if I make `defunctCount` an `uint` as suggested in https://github.com/cri-o/cri-o/pull/5082#pullrequestreview-710249028, I cannot return -1 here. Should I return a 0? But that would create a confusion between whether there are 0 defunct processes or whether there is an error, right?

Reply:
If there is an `err != nil` returned, then we will not use the other result (`defunctCount`) in any case. We can only guarantee that the returned result is correct if `err == nil`, which means we can safely `return 0, err` here :)